gapic/templates/%namespace/%name_%version/%sub/services/%service/transports/rest_asyncio.py.j2

```diff
@@ -3,3 +3,81 @@
  # See related issue: https://github.com/googleapis/gapic-generator-python/issues/2121.
 #}
 {% set rest_async_io_enabled = api.all_library_settings[api.naming.proto_package].python_settings.experimental_features.rest_async_io_enabled %}
+{% extends '_base.py.j2' %}
+
+{% block content %}
+
+from google.api_core import gapic_v1
+
+from typing import Any, Optional
+
+from .rest_base import _Base{{ service.name }}RestTransport
+
+from .base import DEFAULT_CLIENT_INFO as BASE_DEFAULT_CLIENT_INFO
+
+{# TODO (ohmayr): determine the version of rest. aiohttp or google.auth.aio. #}
+DEFAULT_CLIENT_INFO = gapic_v1.client_info.ClientInfo(
+    gapic_version=BASE_DEFAULT_CLIENT_INFO.gapic_version,
+    grpc_version=None,
+    rest_version=None,
+)
+
+class Async{{service.name}}RestTransport(_Base{{ service.name }}RestTransport):
+    """Asynchronous REST backend transport for {{ service.name }}.
+
+    {{ service.meta.doc|rst(width=72, indent=4) }}
+
+    This class defines the same methods as the primary client, so the
+    primary client can load the underlying transport implementation
+    and call it.
+
+    It sends JSON representations of protocol buffers over HTTP/1.1
+    """
+    def __init__(self, *,
```

Comment:
This has fewer parameters than the sync version in `rest.py.j2`. I was assuming the two would look very similar....

Is the plan to add the extra parameters once the auth part is ready?

Reply:
Not all params are supported at the moment.

We will raise an exception for any configured parameter in the client which is not supported in async rest. This will be addressed in a follow up.